.changeset/pink-ducks-fail.md

```diff
@@ -0,0 +1,6 @@
+---
+"@apollo/query-planner": patch
+---
+
+removeInputsFromSelection is tasked with removing inputs to a FetchGroup from the fetch group so that we don't retrieve the same values twice (and FetchGroups that don't functionally do anything can be removed). There was a bug in the code that meant that in certain cases, things that were required were actually being removed. This could manifest in subgraph jumps not being possible because the key it was using to make the jump was not available.
```

Comment:
This is nbd, but the details of the fix are probably less interesting to changelog readers than a description of the bug itself. I might just reorder this to be something like:
"Fix a bug where...xyz." then followed up with implementation details that might explain why it happened.